### DISCUSSION TITLE
Install files to different partition/image

### DISCUSSION THREAD

**Discussion by h0l0gr4m** [OE4T/meta-tegra#1746] [Q&A]
Hi everyone, 

This is a follow up question to this question [Readonly RootFS with an OverlayFS](https://github.com/OE4T/meta-tegra/discussions/1616) but a slightly different topic therefore i opening up a new question.  Btw, thank you everyone for the help there!  
So during the above mentioned question, i learned how to modify the partition layout, and how to tell meta-tegra (or the yocto enviroment) to use my custom partition layout and to create an *.img file ( .e.g overlay.img).   

Now my follow up question is the following: How to  add a new partition , however in contrast to the overlay partition this one should not start empty. That means i have recipes where i want to install to the specific partition / specific image.   

Previously i created the overlay.img with this code : 

```   
    dd if=/dev/zero of=overlay.img bs=1M count=10
    /sbin/mkfs.ext4 overlay.img
```

However from my understanding that creates just an empty image, which this time is not very useful. 

In other words what i wish i know:   
 - How to create a non-empty img file? 
 - How to install to the new partition? 
 
 
 In my head it goes something like this: 
  - Create a new dir: `install -d ${D}/new_partition`
  - Recipes isntall to this new folder
  - Rootfs ignores new_partition folder when creating ext4
  - a new command will generate new img file from the folder new_partition
 
 
 Does this make sense?  Any hints, tips or just links that points me in the right directions will be appreciated. There should be at least one example already in the meta-tegra layer is assume, e.g /boot ? But i could not find it.

**Comment by dwalkes** [comment]
There are some threads linked from https://github.com/OE4T/meta-tegra/discussions/636 about this.  Mender includes a `dataimg` file with tegra which you could possibly use as one example.

**Reply by h0l0gr4m** [reply]
Thank you for your fast answer. Will look into it.
